packages/client/lib/service/ethereumservice.ts

```diff
@@ -80,9 +98,35 @@ export class EthereumService extends Service {
     }
     await super.start()
     void this.synchronizer?.start()
+
+    if (this.v8Engine === null) {
+      this.v8Engine = await getV8Engine()
+    }
+
+    this._statsInterval = setInterval(
+      // eslint-disable-next-line @typescript-eslint/await-thenable
+      await this.stats.bind(this),
+      this.STATS_INTERVAL
+    )
+
     return true
   }
 
+  stats() {
+    if (this.v8Engine !== null) {
+      const { used_heap_size, heap_size_limit } = this.v8Engine.getHeapStatistics()
+
+      const heapUsed = Math.round(used_heap_size / 1000 / 1000) // MB
+      const percentage = Math.round((100 * used_heap_size) / heap_size_limit)
+      this.config.logger.info(`Memory stats usage=${heapUsed} MB percentage=${percentage}%`)
+
+      if (percentage >= this.MEMORY_SHUTDOWN_THRESHOLD && !this.config.shutdown) {
+        this.config.logger.error('EMERGENCY SHUTDOWN DUE TO HIGH MEMORY LOAD...')
```

Comment:
This is a nit but this feels a little shouty.  Can we use normal casing and not all capitals?

Reply:
i think the reason calls for being shouty lol

Reply:
Also like it TBH